lib/portlayer/exec/container_cache.go

```diff
@@ -32,45 +35,41 @@ type containerCache struct {
 	cache map[string]*Container
 }
 
-var containers *containerCache
+var Containers *containerCache
 
 func NewContainerCache() {
 	// cache by the container ID and the vsphere
 	// managed object reference
-	containers = &containerCache{
+	Containers = &containerCache{
 		cache: make(map[string]*Container),
 	}
-
 }
 
 func (conCache *containerCache) Container(idOrRef string) *Container {
 	conCache.m.RLock()
 	defer conCache.m.RUnlock()
 	// find by id or moref
-	container := conCache.cache[idOrRef]
-	return container
+	return conCache.cache[idOrRef]
```

Comment:
I think this is another bug which this PR didn't implement, but still exists.

The map is a map of strings to pointers.  You're returning a reference to a value in the map which can be mutated outside of the cache (or this lock).  Is that intentional?